src/commands/router.ts

```diff
@@ -1,76 +1,106 @@
-import { Message, MessageEmbed } from 'discord.js';
+import { Interaction, Message, MessageEmbed, GuildMember } from 'discord.js';
 import config from '../config';
-import { getMissingPermissions, isAdmin } from '../utils';
+import { isAdmin } from '../utils';
 import CommandError from './CommandError';
 import commands from './instances';
-import * as parser from './parser';
+import { COUNTRIES } from '../utils/countries';
+import { ALL_METRICS } from '../utils';
 import { customCommands } from './CustomCommands';
-import { CustomCommand } from '../types';
 
-export function onError(message: Message, title: string, tip?: string): void {
-  const response = new MessageEmbed().setColor(config.visuals.red).setDescription(title);
-  message.channel.send(tip ? response.setFooter(tip) : response);
-}
-
-export async function onMessageReceived(message: Message): Promise<void> {
-  if (!parser.isValid(message)) {
+export function onError(options: {
+  message?: Message;
+  interaction?: Interaction;
+  title: string;
+  tip?: string;
+}): void {
+  const response = new MessageEmbed().setColor(config.visuals.red).setDescription(options.title);
+  response.setFooter({ text: options.tip ? options.tip : '' });
+  if (options.message) {
+    options.message.channel.send({ embeds: [response] });
     return;
+  } else if (options.interaction && options.interaction.isCommand()) {
+    options.interaction.reply({ embeds: [response] });
```

Comment:
`.reply` won't work if the interaction has been `deferred` iirc, you should use `.followUp` as a blanket solution: This will either `reply`, `editReply`, or add a new reply depending on the interaction status.